app/imageswap-init/imageswap-init.py

```diff
@@ -129,27 +127,44 @@ def build_k8s_csr(namespace, service_name, key):
     csr = x509.CertificateSigningRequestBuilder()
     csr = csr.subject_name(
         # Provide Common Name
-        x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, dns_names[2])])
+        x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, "system:node:" + dns_names[2]), x509.NameAttribute(NameOID.ORGANIZATION_NAME, "system:nodes")])
     )
 
-    csr = csr.add_extension(x509.SubjectAlternativeName([x509.DNSName(dns_names[0]), x509.DNSName(dns_names[1]), x509.DNSName(dns_names[2]),]), critical=False,)
+    csr = csr.add_extension(
+        x509.SubjectAlternativeName(
+            [
+                x509.DNSName(dns_names[0]),
+                x509.DNSName(dns_names[1]),
+                x509.DNSName(dns_names[2]),
+            ]
+        ),
+        critical=False,
+    )
 
     # Sign the CSR with our private key.
     csr = csr.sign(key, hashes.SHA256(), default_backend())
 
     csr_pem = csr.public_bytes(serialization.Encoding.PEM)
 
     # Build Kubernetes CSR
-    k8s_csr_meta = client.V1ObjectMeta(name=dns_names[1] + ".cert-request", namespace=namespace, labels={"app": "imageswap"},)
+    k8s_csr_meta = client.V1ObjectMeta(
+        name=dns_names[1] + ".cert-request",
+        namespace=namespace,
+        labels={"app": "imageswap"},
+    )
 
-    k8s_csr_spec = client.V1beta1CertificateSigningRequestSpec(
-        groups=["system:authenticated"],
-        usages=["digital signature", "key encipherment", "server auth"],
+    k8s_csr_spec = client.V1CertificateSigningRequestSpec(
+        groups=["system:nodes"],
```

Comment:
Is this needed? I couldn't find a reference to needing this in the k8s docs. If you have a link, please drop it here

Reply:
The only reference I have here is [Certificate Signing Requests](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/). And while I was trying to make things work I found: 

`status:
  conditions:
  ...
    reason: ImageSwap-Approve
    status: "True"
    type: Approved
  - lastTransitionTime: ""
    message: subject organization is not system:nodes
    reason: SignerValidationFailure
    status: "True"
    type: Failed`
[Pipeline execution](https://github.com/mariomcosta/imageswap-webhook/runs/6218265476?check_suite_focus=true)

So it was like trying with some setup, looking at the logs, and making some changes to run.